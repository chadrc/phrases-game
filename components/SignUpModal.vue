<template>
<div class="modal" :class="{'is-active': signingUp}">
  <div class="modal-background"></div>
  <div class="modal-card">
    <header class="modal-card-head">
      <p class="modal-card-title">Sign Up</p>
      <button class="delete" aria-label="close" @click="endSignUp()"></button>
    </header>
    <section class="modal-card-body">
      <!-- Content ... -->
    </section>
    <footer class="modal-card-foot">
      <button class="button is-success">Sign Up</button>
      <button class="button" @click="endSignUp()">Cancel</button>
    </footer>
  </div>
</div>
</template>

<script>
import { mapGetters, mapActions } from "vuex";

export default {
  computed: {
    ...mapGetters("auth", ["signingUp"])
  },
  methods: {
    ...mapActions("auth", ["endSignUp"])
  }
};
</script>

<style scoped>
</style>
