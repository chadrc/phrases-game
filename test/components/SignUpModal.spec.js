import SignUpModal from "@/components/SignUpModal.vue";
import { makeVueMock } from "@/test/utils";

describe("SignUpModal", () => {
  const { wrapper, store } = makeVueMock(SignUpModal);

  test("is vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders regular", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("renders active", () => {
    store.commit("auth/setSigningUp", true);
    expect(wrapper.element).toMatchSnapshot();
  });
});
