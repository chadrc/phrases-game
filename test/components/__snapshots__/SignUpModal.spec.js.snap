// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpModal renders active 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Sign Up
      </p>
       
      <button
        aria-label="close"
        class="delete"
      />
    </header>
     
    <section
      class="modal-card-body"
    />
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button is-success"
      >
        Sign Up
      </button>
       
      <button
        class="button"
      >
        Cancel
      </button>
    </footer>
  </div>
</div>
`;

exports[`SignUpModal renders regular 1`] = `
<div
  class="modal"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Sign Up
      </p>
       
      <button
        aria-label="close"
        class="delete"
      />
    </header>
     
    <section
      class="modal-card-body"
    />
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button is-success"
      >
        Sign Up
      </button>
       
      <button
        class="button"
      >
        Cancel
      </button>
    </footer>
  </div>
</div>
`;
